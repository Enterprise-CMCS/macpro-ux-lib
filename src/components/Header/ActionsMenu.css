.user-actions-drop-down li {
  cursor: pointer;
  margin-bottom: 10px;
}

.user-actions-drop-down li a {
  align-items: center;
  display: flex;
  flex-direction: row;
}
.user-actions-drop-down li a,
.menu-label {
  font-size: 1.06rem;
}

@media only screen and (min-width: 64em) {
  .user-actions-menu {
    color: #fff;
  }

  .user-actions-menu .menu-label {
    align-items: center;
    cursor: pointer;
    display: flex;
    flex-direction: row;
    font-weight: bold;
    position: relative;
  }

  .user-actions-drop-down {
    background-color: #0071bc;
    flex-direction: flex-start;
    left: 25px;
    position: absolute;
    top: 40px;
    width: 250px;
  }

  .user-actions-drop-down li {
    padding: 5px 15px 5px;
  }

  .user-actions-drop-down li a {
    color: #fff;
  }

  .user-actions-drop-down li a svg {
    color: #fff;
    margin-right: 10px;
  }

  .user-actions-drop-down li a:hover {
    color: #fff;
    text-decoration: unset;
  }

  .user-actions-drop-down {
    display: none;
    padding-top: 20px;
  }
}

@media only screen and (max-width: 64em) {
  .user-actions-menu {
    padding: 0;
  }
  .user-actions-menu a {
    padding: 12px 16px;
  }
  .user-actions-drop-down {
    display: block;
  }

  .user-actions-drop-down li {
    margin: 0;
    border-top: 1px solid #dfe1e2;
  }

  .user-actions-drop-down li a:hover,
  .user-actions-drop-down li a:hover svg {
    color: #005ea2;
    background-color: #f0f0f0;
    text-decoration: none;
  }

  .user-actions-drop-down li a svg {
    color: #565c65;
  }
}
