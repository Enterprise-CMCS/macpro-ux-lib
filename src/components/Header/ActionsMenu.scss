@use "../../assets/theme/macbis-colors" as c;

.user-actions-container {
  .user-actions-menu li {
    cursor: pointer;
  }

  .user-actions-menu a {
    align-items: center;
    display: flex;
    flex-direction: row;
  }
  .user-actions-menu a,
  .menu-label {
    font-size: 1.06rem;
  }

  .user-actions-menu svg {
    margin-right: 10px;
  }

  @media only screen and (min-width: 64em) {
    // USWDS Overrides
    .usa-nav__submenu,
    .usa-header--extended .usa-megamenu.usa-nav__submenu::after,
    .usa-header--extended .usa-megamenu.usa-nav__submenu::before {
      background-color: c.$primary-darkest;
    }

    .usa-nav__primary button {
      color: c.$gray;
    }

    .usa-header--extended .usa-nav__primary-item > .usa-current::after {
      background-color: c.$primary;
    }

    // New Component Styles
    .menu-label {
      color: #fff;
      align-items: center;
      cursor: pointer;
      display: flex;
      flex-direction: row;
      font-weight: bold;
      position: relative;
    }

    .user-actions-menu {
      display: none;
      background-color: c.$primary;
      flex-direction: flex-start;
      left: 25px;
      position: absolute;
      top: 41px;
      width: 250px;
    }

    .user-actions-menu a {
      padding: 10px 15px 10px;
    }

    .user-actions-menu {
      a,
      svg,
      a:hover {
        color: #fff;
        text-decoration: unset;
      }
    }

    .user-actions-menu a:hover,
    .user-actions-menu li:hover {
      background-color: c.$primary-darker;
    }
  }

  @media only screen and (max-width: 64em) {
    // USWDS Overrides
    .usa-menu-btn {
      background-color: c.$primary;
    }
    .usa-menu-btn:hover {
      background-color: c.$primary-darker;
    }

    .usa-nav__primary .usa-current,
    .usa-nav__primary .usa-current::after {
      color: c.$primary-darker;
    }

    .usa-nav__primary button {
      color: c.$gray;
    }

    .usa-nav__primary button:hover {
      color: c.$primary-darker;
      background-color: c.$gray-lightest;
    }

    .usa-nav__submenu-item {
      border-top: 1px solid c.$gray-lighter;
    }
    .usa-nav__primary a:not(.usa-button):not(.usa-current) {
      color: c.$gray;
    }

    .usa-nav__primary a:not(.usa-button):not(.usa-current):hover {
      color: c.$primary-darker;
    }

    // New Component Styles
    padding: 0;

    a {
      padding: 12px 16px;
    }

    .user-actions-menu {
      display: block;
    }

    .user-actions-menu,
    .user-actions-menu svg {
      color: c.$gray;
    }

    .user-actions-menu li {
      border-top: 1px solid c.$gray-lighter;
    }

    .user-actions-menu a:hover,
    .user-actions-menu a:hover svg {
      color: c.$primary-darker;
      background-color: c.$gray-lightest;
      text-decoration: none;
    }
  }
}
