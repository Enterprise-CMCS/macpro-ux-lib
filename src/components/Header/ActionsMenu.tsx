import React from "react";
import { Icon, IconChoice } from "components/Icon/Icon";

interface Link {
  href?: string;
  iconName?: IconChoice;
  onClick?: () => any;
  text: string;
}

interface Props {
  name: string;
  links: Link[];
}

/**
 * ActionsMenu Component
 *
 * @param {string} name   The name of the menu.
 * @param {Link[]} links  List of links to display in the menu.
 */

export const ActionsMenu: React.FC<Props> = ({ name, links, ...rest }) => {
  const [isOpen, setIsOpen] = React.useState(false);

  return (
    <div className="user-actions-container">
      <span className="menu-label" onClick={() => setIsOpen(!isOpen)}>
        {name}
        <Icon
          aria-hidden="true"
          name={isOpen ? "arrow_drop_up" : "arrow_drop_down"}
          color="#fff"
        />
      </span>
      <menu
        className="user-actions-menu usa-nav__secondary-links"
        style={isOpen ? { display: "block" } : {}}
        {...rest}
      >
        {links.map((link, idx) => {
          return (
            <li key={idx}>
              <a href={link.href} onClick={link.onClick}>
                {link.iconName && <Icon name={link.iconName} color="#fff" />}
                {link.text}
              </a>
            </li>
          );
        })}
      </menu>
    </div>
  );
};
