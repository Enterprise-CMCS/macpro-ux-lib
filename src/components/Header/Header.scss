@use "../../sass/macbis-colors" as c;

.usa-header {
  @media only screen and (min-width: 1024px) {
    .usa-nav {
      border: none;
    }

    .usa-navbar-container {
      background-color: c.$primary;
      min-height: 100px;
      .usa-navbar {
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        height: 100px;
        align-items: center;
        overflow: visible;
        .usa-nav__secondary {
          bottom: unset;
          margin: 0;
          position: relative;
          z-index: 999;
        }
      }
    }

    .usa-logo .application-logo-container {
      color: #fff;
    }
  }

  @media only screen and (max-width: 1023px) {
    .usa-navbar-container {
      background-color: c.$primary;
      min-height: 70px;
      .usa-navbar {
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        height: 75px;
        align-items: center;
        overflow: visible;
        .usa-nav__secondary {
          bottom: unset;
          margin: 0;
          position: relative;
          z-index: 999;
        }
      }
      button {
        margin-right: 10px;
      }
    }
    .usa-logo .application-logo-container {
      color: c.$primary;
    }
  }
}
