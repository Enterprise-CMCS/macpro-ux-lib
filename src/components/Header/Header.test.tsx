import React from "react";
import { fireEvent, screen, render } from "../../test-setup";
import { ActionsMenu } from "./ActionsMenu";

import { Header } from "./Header";
import { Link } from "../Link/Link";
import { Logo } from "../Logo/Logo";

const testOnClick = jest.fn();

let navData: {
  buttonText: string;
  current?: boolean;
  columns: { text: string; href?: string; onClick?: () => any }[][];
}[];
const setNavData = () => {
  navData = [
    {
      buttonText: "Current Section",
      current: true,
      columns: [
        [
          { text: "Navigational Link", href: "", onClick: testOnClick },
          { text: "Navigational Link", href: "" },
          { text: "Navigational Link", href: "" },
          { text: "Navigational Link", href: "" },
        ],
        [{ text: "Navigational Link", href: "" }],
        [
          { text: "Navigational Link", href: "" },
          { text: "Navigational Link", href: "" },
          { text: "Navigational Link", href: "" },
        ],
        [
          { text: "Navigational Link", href: "" },
          { text: "Navigational Link", href: "" },
        ],
      ],
    },
    {
      buttonText: "Section",
      columns: [
        [
          { text: "Navigational Link", href: "" },
          { text: "Navigational Link", href: "" },
          { text: "Navigational Link", href: "" },
          { text: "Navigational Link", href: "" },
          { text: "Navigational Link", href: "" },
          { text: "Navigational Link", href: "" },
          { text: "Navigational Link", href: "" },
        ],
      ],
    },
  ];
};

const actionsMenuLinks = [
  {
    text: "Manage Profile",
    iconName: "person",
    onClick: () => console.log("Manage Profile"),
  },
  {
    text: "Request Role Change",
    iconName: "people",
    onClick: () => console.log("Request Role Change"),
  },
  {
    text: "Log Out",
    iconName: "logout",
    onClick: () => console.log("Log Out"),
  },
];

describe("Test the Header component", () => {
  beforeEach(() => {
    setNavData();
  });

  it("should render correctly", () => {
    render(
      <Header
        navData={navData}
        headerLogo={
          <Link href="" title="Project Title">
            <Logo altText={"CMS.gov Project"} />
          </Link>
        }
      />
    );

    // All text should be rendered
    expect(screen.getByText("CMS.gov Project")).toBeInTheDocument();
    expect(screen.getByText("Current Section")).toBeInTheDocument();
    expect(screen.getByText("Section")).toBeInTheDocument();

    // All links should be rendered
    expect(screen.getAllByText("Navigational Link").length).toBe(17);
    const link = screen.getAllByText("Navigational Link")[0];
    expect(link).toHaveAttribute("href");
    fireEvent.click(link);
    expect(testOnClick).toHaveBeenCalled();
  });

  it("nav items should expand and collapse on click", () => {
    const { container } = render(
      <Header
        navData={navData}
        headerLogo={
          <Link href="" title="Project Title">
            <Logo altText={"CMS.gov Project"} />
          </Link>
        }
      />
    );
    const currentSection = screen.getByText("Current Section");
    const navSection = container.getElementsByClassName("usa-nav__submenu")[0]; // assumes 1st

    // Assert collapsed
    expect(navSection).toHaveAttribute("hidden");

    fireEvent.click(currentSection);

    // Assert expanded
    expect(navSection).not.toHaveAttribute("hidden");

    fireEvent.click(currentSection);

    // Assert collapsed
    expect(navSection).toHaveAttribute("hidden");
  });

  it("nav items should collapse on outside click", () => {
    const { container } = render(
      <Header
        navData={navData}
        headerLogo={
          <Link href="" title="Project Title">
            <Logo altText={"CMS.gov Project"} />
          </Link>
        }
      />
    );
    const body = document.body;
    const currentSection = screen.getByText("Current Section");
    const navSection = container.getElementsByClassName("usa-nav__submenu")[0]; // assumes 1st

    // Assert collapsed
    expect(navSection).toHaveAttribute("hidden");

    fireEvent.click(currentSection);

    // Assert expanded
    expect(navSection).not.toHaveAttribute("hidden");

    fireEvent.mouseDown(body);

    // Assert collapsed
    expect(navSection).toHaveAttribute("hidden");
  });

  it("should render correcty in mobile view", () => {
    const { container } = render(
      <Header
        navData={navData}
        headerLogo={
          <Link href="" title="Project Title">
            <Logo altText={"CMS.gov Project"} />
          </Link>
        }
      />
    );
    const menuBtn = screen.getByText("Menu");
    const closeBtn = container.getElementsByClassName("usa-nav__close")[0];
    const body = document.body;
    const nav = container.getElementsByClassName("usa-nav")[0];
    const overlay = container.getElementsByClassName("usa-overlay")[0];

    // Assert appropriate default classes
    expect(body.className).toBe("");
    expect(nav.className).toBe("usa-nav");
    expect(overlay.className).toBe("usa-overlay");

    fireEvent.click(menuBtn);

    // Assert "is-visible" classes added
    expect(body.className).toBe("usa-js-mobile-nav--active");
    expect(nav.className).toBe("usa-nav is-visible");
    expect(overlay.className).toBe("usa-overlay is-visible");

    fireEvent.click(closeBtn);

    // Assert "is-visible" classes removed
    expect(body.className).toBe("");
    expect(nav.className).toBe("usa-nav");
    expect(overlay.className).toBe("usa-overlay");
  });

  it("in mobile view, clicking outside the menu should trigger close", () => {
    const { container } = render(
      <Header
        navData={navData}
        headerLogo={
          <Link href="" title="Project Title">
            <Logo altText={"CMS.gov Project"} />
          </Link>
        }
      />
    );
    const menuBtn = screen.getByText("Menu");
    const body = document.body;
    const nav = container.getElementsByClassName("usa-nav")[0];
    const overlay = container.getElementsByClassName("usa-overlay")[0];

    // Assert appropriate default classes
    expect(body.className).toBe("");
    expect(nav.className).toBe("usa-nav");
    expect(overlay.className).toBe("usa-overlay");

    fireEvent.click(menuBtn);

    // Assert "is-visible" classes added
    expect(body.className).toBe("usa-js-mobile-nav--active");
    expect(nav.className).toBe("usa-nav is-visible");
    expect(overlay.className).toBe("usa-overlay is-visible");

    fireEvent.mouseDown(body); // outside click

    // Assert "is-visible" classes removed
    expect(body.className).toBe("");
    expect(nav.className).toBe("usa-nav");
    expect(overlay.className).toBe("usa-overlay");
  });

  describe("current prop should add an underline", () => {
    // current should control the active state of the button
    it("should exist on Current Section", () => {
      const { container } = render(
        <Header
          navData={navData}
          headerLogo={
            <Link href="" title="Project Title">
              <Logo altText={"CMS.gov Project"} />
            </Link>
          }
        />
      );
      expect(container.getElementsByClassName("usa-current").length).toBe(1);
      const button = container.getElementsByClassName("usa-current")[0];
      expect(button.innerHTML).toContain("Current Section");
    });

    it("should not exist on Current Section", () => {
      // remove current prop from navData
      navData[0].current = false;
      const { container } = render(
        <Header
          navData={navData}
          headerLogo={
            <Link href="" title="Project Title">
              <Logo altText={"CMS.gov Project"} />
            </Link>
          }
        />
      );
      expect(container.getElementsByClassName("usa-current").length).toBe(0);
    });
  });

  describe("component snapshots", () => {
    it("default", () => {
      const { container } = render(
        <Header
          headerLogo={
            <Link href="" title="Project Title">
              <Logo altText={"CMS.gov Project"} />
            </Link>
          }
        />
      );
      expect(container).toMatchSnapshot();
    });

    it("with navigation", () => {
      const { container } = render(
        <Header
          navData={navData}
          headerLogo={
            <Link href="" title="Project Title">
              <Logo altText={"CMS.gov Project"} />
            </Link>
          }
        />
      );
      expect(container).toMatchSnapshot();
    });

    it("with ActionsMenu", () => {
      const { container } = render(
        <Header
          navData={navData}
          headerLogo={
            <Link href="" title="Project Title">
              <Logo altText={"CMS.gov Project"} />
            </Link>
          }
          secondaryComponent={
            <ActionsMenu name="My Account" links={actionsMenuLinks} />
          }
        />
      );
      expect(container).toMatchSnapshot();
    });
  });
});
