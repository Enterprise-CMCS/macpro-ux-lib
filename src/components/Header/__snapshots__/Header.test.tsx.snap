// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Header component component snapshots default 1`] = `
<div>
  <div
    class="usa-overlay"
  />
  <header
    class="usa-header usa-header--extended"
  >
    <div
      class="usa-navbar-container"
    >
      <div
        class="usa-navbar"
      >
        <div
          class="usa-logo"
          id="basic-logo"
        >
          <a
            class="usa-link"
            href=""
            style="cursor: pointer;"
            title="Project Title"
          >
            <div
              aria-label="CMS.gov Project"
              class="application-logo-container "
              role="img"
            >
              <h2
                class="usa-logo__text"
              >
                CMS.gov Project
              </h2>
            </div>
          </a>
        </div>
        <button
          aria-label="Menu button"
          class="usa-menu-btn"
          formtarget="_self"
        >
          <span
            class="padding-x-05"
          >
            Menu
          </span>
        </button>
      </div>
    </div>
    <nav
      aria-label="Primary navigation"
      class="usa-nav"
    >
      <div
        class="usa-nav__inner"
      >
        <button
          aria-label=" button"
          class="usa-nav__close"
          formtarget="_self"
        >
          <span
            class="padding-x-05 padding-top-05"
          />
          <div>
            <svg
              aria-hidden="true"
              aria-label="close icon"
              class="usa-icon--size-3"
              color="curentColor"
              fill="currentColor"
              role="img"
            >
              <use
                href="sprite.svg#close"
              />
            </svg>
          </div>
        </button>
        <ul
          class="usa-nav__primary usa-accordion"
        />
        <div
          class="usa-nav__secondary usa-header--extended"
        />
      </div>
    </nav>
  </header>
</div>
`;

exports[`Test the Header component component snapshots with ActionsMenu 1`] = `
<div>
  <div
    class="usa-overlay"
  />
  <header
    class="usa-header usa-header--extended"
  >
    <div
      class="usa-navbar-container"
    >
      <div
        class="usa-navbar"
      >
        <div
          class="usa-logo"
          id="basic-logo"
        >
          <a
            class="usa-link"
            href=""
            style="cursor: pointer;"
            title="Project Title"
          >
            <div
              aria-label="CMS.gov Project"
              class="application-logo-container "
              role="img"
            >
              <h2
                class="usa-logo__text"
              >
                CMS.gov Project
              </h2>
            </div>
          </a>
        </div>
        <button
          aria-label="Menu button"
          class="usa-menu-btn"
          formtarget="_self"
        >
          <span
            class="padding-x-05"
          >
            Menu
          </span>
        </button>
      </div>
    </div>
    <nav
      aria-label="Primary navigation"
      class="usa-nav"
    >
      <div
        class="usa-nav__inner"
      >
        <button
          aria-label=" button"
          class="usa-nav__close"
          formtarget="_self"
        >
          <span
            class="padding-x-05 padding-top-05"
          />
          <div>
            <svg
              aria-hidden="true"
              aria-label="close icon"
              class="usa-icon--size-3"
              color="curentColor"
              fill="currentColor"
              role="img"
            >
              <use
                href="sprite.svg#close"
              />
            </svg>
          </div>
        </button>
        <ul
          class="usa-nav__primary usa-accordion"
        >
          <li
            class="usa-nav__primary-item"
          >
            <button
              aria-controls="extended-mega-nav-section-0"
              aria-expanded="false"
              aria-label="Current Section button"
              class="usa-accordion__button usa-nav__link usa-current"
              formtarget="_self"
            >
              <span
                class="padding-x-05"
              >
                Current Section
              </span>
            </button>
            <div
              class="usa-nav__submenu usa-megamenu"
              hidden=""
              id="extended-mega-nav-section-0"
            >
              <div
                class="grid-row grid-gap-4"
              >
                <div
                  class="usa-col"
                >
                  <ul
                    class="usa-nav__submenu-list"
                  >
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                  </ul>
                </div>
                <div
                  class="usa-col"
                >
                  <ul
                    class="usa-nav__submenu-list"
                  >
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                  </ul>
                </div>
                <div
                  class="usa-col"
                >
                  <ul
                    class="usa-nav__submenu-list"
                  >
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                  </ul>
                </div>
                <div
                  class="usa-col"
                >
                  <ul
                    class="usa-nav__submenu-list"
                  >
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </li>
          <li
            class="usa-nav__primary-item"
          >
            <button
              aria-controls="extended-mega-nav-section-1"
              aria-expanded="false"
              aria-label="Section button"
              class="usa-accordion__button usa-nav__link "
              formtarget="_self"
            >
              <span
                class="padding-x-05"
              >
                Section
              </span>
            </button>
            <div
              class="usa-nav__submenu usa-megamenu"
              hidden=""
              id="extended-mega-nav-section-1"
            >
              <div
                class="grid-row grid-gap-4"
              >
                <div
                  class="usa-col"
                >
                  <ul
                    class="usa-nav__submenu-list"
                  >
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </li>
        </ul>
        <div
          class="usa-nav__secondary usa-header--extended"
        >
          <div
            class="user-actions-container"
          >
            <span
              class="menu-label"
            >
              My Account
              <svg
                aria-hidden="true"
                aria-label="arrow drop down icon"
                class="usa-icon--size-3"
                color="#fff"
                fill="currentColor"
                role="img"
              >
                <use
                  href="sprite.svg#arrow_drop_down"
                />
              </svg>
            </span>
            <menu
              class="user-actions-menu usa-nav__secondary-links"
            >
              <li>
                <a
                  class="usa-link"
                  style="cursor: pointer;"
                  title="Manage Profile Link"
                >
                  <svg
                    aria-hidden="true"
                    aria-label="person icon"
                    class="usa-icon--size-3"
                    color="black"
                    fill="currentColor"
                    role="img"
                  >
                    <use
                      href="sprite.svg#person"
                    />
                  </svg>
                  Manage Profile
                </a>
              </li>
              <li>
                <a
                  class="usa-link"
                  style="cursor: pointer;"
                  title="Request Role Change Link"
                >
                  <svg
                    aria-hidden="true"
                    aria-label="people icon"
                    class="usa-icon--size-3"
                    color="black"
                    fill="currentColor"
                    role="img"
                  >
                    <use
                      href="sprite.svg#people"
                    />
                  </svg>
                  Request Role Change
                </a>
              </li>
              <li>
                <a
                  class="usa-link"
                  style="cursor: pointer;"
                  title="Log Out Link"
                >
                  <svg
                    aria-hidden="true"
                    aria-label="logout icon"
                    class="usa-icon--size-3"
                    color="black"
                    fill="currentColor"
                    role="img"
                  >
                    <use
                      href="sprite.svg#logout"
                    />
                  </svg>
                  Log Out
                </a>
              </li>
            </menu>
          </div>
        </div>
      </div>
    </nav>
  </header>
</div>
`;

exports[`Test the Header component component snapshots with navigation 1`] = `
<div>
  <div
    class="usa-overlay"
  />
  <header
    class="usa-header usa-header--extended"
  >
    <div
      class="usa-navbar-container"
    >
      <div
        class="usa-navbar"
      >
        <div
          class="usa-logo"
          id="basic-logo"
        >
          <a
            class="usa-link"
            href=""
            style="cursor: pointer;"
            title="Project Title"
          >
            <div
              aria-label="CMS.gov Project"
              class="application-logo-container "
              role="img"
            >
              <h2
                class="usa-logo__text"
              >
                CMS.gov Project
              </h2>
            </div>
          </a>
        </div>
        <button
          aria-label="Menu button"
          class="usa-menu-btn"
          formtarget="_self"
        >
          <span
            class="padding-x-05"
          >
            Menu
          </span>
        </button>
      </div>
    </div>
    <nav
      aria-label="Primary navigation"
      class="usa-nav"
    >
      <div
        class="usa-nav__inner"
      >
        <button
          aria-label=" button"
          class="usa-nav__close"
          formtarget="_self"
        >
          <span
            class="padding-x-05 padding-top-05"
          />
          <div>
            <svg
              aria-hidden="true"
              aria-label="close icon"
              class="usa-icon--size-3"
              color="curentColor"
              fill="currentColor"
              role="img"
            >
              <use
                href="sprite.svg#close"
              />
            </svg>
          </div>
        </button>
        <ul
          class="usa-nav__primary usa-accordion"
        >
          <li
            class="usa-nav__primary-item"
          >
            <button
              aria-controls="extended-mega-nav-section-0"
              aria-expanded="false"
              aria-label="Current Section button"
              class="usa-accordion__button usa-nav__link usa-current"
              formtarget="_self"
            >
              <span
                class="padding-x-05"
              >
                Current Section
              </span>
            </button>
            <div
              class="usa-nav__submenu usa-megamenu"
              hidden=""
              id="extended-mega-nav-section-0"
            >
              <div
                class="grid-row grid-gap-4"
              >
                <div
                  class="usa-col"
                >
                  <ul
                    class="usa-nav__submenu-list"
                  >
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                  </ul>
                </div>
                <div
                  class="usa-col"
                >
                  <ul
                    class="usa-nav__submenu-list"
                  >
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                  </ul>
                </div>
                <div
                  class="usa-col"
                >
                  <ul
                    class="usa-nav__submenu-list"
                  >
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                  </ul>
                </div>
                <div
                  class="usa-col"
                >
                  <ul
                    class="usa-nav__submenu-list"
                  >
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </li>
          <li
            class="usa-nav__primary-item"
          >
            <button
              aria-controls="extended-mega-nav-section-1"
              aria-expanded="false"
              aria-label="Section button"
              class="usa-accordion__button usa-nav__link "
              formtarget="_self"
            >
              <span
                class="padding-x-05"
              >
                Section
              </span>
            </button>
            <div
              class="usa-nav__submenu usa-megamenu"
              hidden=""
              id="extended-mega-nav-section-1"
            >
              <div
                class="grid-row grid-gap-4"
              >
                <div
                  class="usa-col"
                >
                  <ul
                    class="usa-nav__submenu-list"
                  >
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                    <li
                      class="usa-nav__submenu-item"
                    >
                      <a
                        class="usa-link"
                        href=""
                        style="cursor: pointer;"
                        title="Navigational Link Link"
                      >
                        Navigational Link
                      </a>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </li>
        </ul>
        <div
          class="usa-nav__secondary usa-header--extended"
        />
      </div>
    </nav>
  </header>
</div>
`;
